{
  "comments": [
    {
      "key": {
        "uuid": "e83ff085_5f205cff",
        "filename": "resources/libraries/python/autogen/Regenerator.py",
        "patchSetId": 2
      },
      "lineNbr": 46,
      "author": {
        "id": 1848
      },
      "writtenOn": "2019-08-16T12:29:03Z",
      "side": 1,
      "message": "I\u0027m new here, but python string objects are immutable.",
      "range": {
        "startLine": 46,
        "startChar": 36,
        "endLine": 46,
        "endChar": 70
      },
      "revId": "851190bd71e3bd77acea99acf63c4935335673d0",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2",
      "unresolved": true
    },
    {
      "key": {
        "uuid": "39033594_07824119",
        "filename": "resources/libraries/python/autogen/Regenerator.py",
        "patchSetId": 2
      },
      "lineNbr": 137,
      "author": {
        "id": 1848
      },
      "writtenOn": "2019-08-16T12:29:03Z",
      "side": 1,
      "message": "given the above comment, the caller should not rely on this and sibling functions to bump the num, but should do it on its own.",
      "range": {
        "startLine": 137,
        "startChar": 11,
        "endLine": 137,
        "endChar": 18
      },
      "revId": "851190bd71e3bd77acea99acf63c4935335673d0",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2",
      "unresolved": true
    },
    {
      "key": {
        "uuid": "ec9a23df_e4e2a92f",
        "filename": "resources/libraries/python/autogen/Regenerator.py",
        "patchSetId": 2
      },
      "lineNbr": 266,
      "author": {
        "id": 1848
      },
      "writtenOn": "2019-08-16T12:29:03Z",
      "side": 1,
      "message": "FYI dicts iterate over their keys already. Unless you modify the dict while iterating, you don\u0027t need to create a copy of the keys.",
      "range": {
        "startLine": 266,
        "startChar": 46,
        "endLine": 266,
        "endChar": 53
      },
      "revId": "851190bd71e3bd77acea99acf63c4935335673d0",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2",
      "unresolved": true
    },
    {
      "key": {
        "uuid": "5dc9225a_29a0e0fc",
        "filename": "resources/libraries/python/autogen/Regenerator.py",
        "patchSetId": 2
      },
      "lineNbr": 343,
      "author": {
        "id": 1848
      },
      "writtenOn": "2019-08-16T12:29:03Z",
      "side": 1,
      "message": "It is always frustrating when a library assumes it will only be called from a specific program, and that it can just decide to terminate its client application at will.\n\nLibraries should raise (relevant) exceptions, and let the application decide what is its exit code, and whether it is to exit at all.",
      "range": {
        "startLine": 343,
        "startChar": 16,
        "endLine": 343,
        "endChar": 27
      },
      "revId": "851190bd71e3bd77acea99acf63c4935335673d0",
      "serverId": "6d2eb258-4fe2-443e-8a38-ca81da23d4c2",
      "unresolved": true
    }
  ]
}